seml/start.py

```diff
@@ -504,16 +619,34 @@ def start_experiments(db_collection_name, local, sacred_id, batch_id, filter_dic
     if sacred_id is None:
         filter_dict = build_filter_dict([], batch_id, filter_dict)
     else:
-        filter_dict = {'_id': sacred_id}
+        # if we have a specific sacred ID, we ignore the state of the experiment and run it in any case.
+        all_states = (States.PENDING + States.STAGED + States.RUNNING + States.FAILED + States.INTERRUPTED
+                      + States.KILLED + States.COMPLETED)
+        filter_dict.update({'_id': sacred_id, "status": {"$in": all_states}})
+
+    collection = get_collection(db_collection_name)
 
     if dry_run:
-        print_commands(db_collection_name, unobserved=unobserved, post_mortem=post_mortem,
+        print_commands(collection, unobserved=unobserved, post_mortem=post_mortem,
                        num_exps=num_exps, filter_dict=filter_dict)
-    else:
-        start_jobs(db_collection_name, slurm=use_slurm,
-                   unobserved=unobserved, post_mortem=post_mortem,
-                   num_exps=num_exps, filter_dict=filter_dict, dry_run=dry_run,
-                   output_to_file=output_to_file)
+        return
+
+    staged_experiments = get_staged_experiments(collection=collection, filter_dict=filter_dict, num_exps=num_exps,
+                                                slurm=use_slurm, set_to_pending=set_to_pending)
+
+    if len(staged_experiments) == 0 and use_slurm:
```

Comment:
Why only when using Slurm? We have a different command for launching workers.

Reply:
Or do you want to merge start and launch_worker?

Reply:
Yes, currently `start_experiments` is called both from `seml start` and `seml launch-worker`.